consensus/core/src/core.rs

```diff
@@ -91,43 +95,51 @@ impl Core {
             threshold_clock: ThresholdClock::new(0, context),
             last_proposed_block: my_genesis_block,
             transaction_consumer,
-            pending_ancestors: BTreeMap::new(),
+            last_included_ancestors: BTreeMap::new(),
             block_manager,
             committer,
             last_decided_leader,
             commit_observer,
             signals,
             block_signer,
+            dag_state,
             store,
         }
-        .recover(all_genesis_blocks)
+        .recover()
     }
 
-    fn recover(mut self, genesis_blocks: Vec<VerifiedBlock>) -> Self {
-        // We always need the genesis blocks as a starter point since we might not have advanced yet at all.
-        let mut all_blocks = genesis_blocks;
-
-        // Now fetch the proposed blocks per authority for their last two rounds.
-        let context = self.context.clone();
-        for (index, _authority) in context.committee.authorities() {
-            let blocks = self
-                .store
-                .scan_last_blocks_by_author(index, 2)
-                .expect("Storage error while recovering Core");
-            all_blocks.extend(blocks);
-        }
+    fn recover(mut self) -> Self {
+        // Fetch the proposed blocks per authority for their last two rounds.
+        let all_blocks = self
+            .context
+            .committee
+            .authorities()
+            .flat_map(|(index, _authority)| {
+                self.store
```

Comment:
should we go dag_state -> store instead of directly accessing the store from here?